### PR TITLE
fix: display instructors on mobile as well

### PR DESCRIPTION
**Pull request by arslanashraf7**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
#4961 

#### What's this PR do?
- Displays the instructors on mobile and tablets as well
- Updates the styling to fix left arrows spacing on mobile/tablet
- Fixes left arrow overlap display issue on mobile

#### How should this be manually tested?
- Add a program page through CMS
- Add instructors within the page
- Load the page and check the instructor's list is showing up fine on mobile/tablets as well as on desktop versions

#### Where should the reviewer start?
Program Page

#### Screenshots
**Desktop Screenshot**
Screenshot 2021-06-23 at 3 52 35 PM

**Mobile Screenshot**
Screenshot 2021-06-23 at 3 52 48 PM

**Mobile Second Item Screenshot**
Screenshot 2021-06-23 at 3 53 01 PM

**Tablet Screenshot**
Screenshot 2021-06-23 at 4 10 46 PM

### CODE REVIEW

**Comment by arslanashraf7** [comment]
@pdpinch, Could you please take a look at the attached screenshots In the PR description and let me know if they are fine by you.

File: static/js/components/FacultyCarousel.js [hunk 1]
```diff
@@ -19,11 +19,15 @@ export default class FacultyCarousel extends React.Component {
...
           breakpoint: 600,
-          settings:   "unslick"
+          settings:   {
+            slidesToShow: 1
+          }
         },
         {
-          breakpoint: 800,
-          settings:   "unslick"
+          breakpoint: 840,
```
Comment (pdpinch) [inline static/js/components/FacultyCarousel.js:27]: Are we standardizing on a tablet breakpoint of 840px?

File: static/js/components/FacultyCarousel.js [hunk 2]
```diff
@@ -19,11 +19,15 @@ export default class FacultyCarousel extends React.Component {
       responsive:     [
         {
           breakpoint: 600,
-          settings:   "unslick"
+          settings:   {
+            slidesToShow: 1
+          }
```
Comment (pdpinch) [inline static/js/components/FacultyCarousel.js:24]: Can you tell me more about this change? Is this because of a change to the slick carousel API?

**Review by pdpinch — COMMENTED** [review]
1 slide for the tablet view is very big. Did you try 2? 

If 1 is the best option, this is OK.

File: static/js/components/FacultyCarousel.js (hunk 2 above)
Comment (arslanashraf7) [reply static/js/components/FacultyCarousel.js:24]: The last update made to `react-slick` package in MM was made around 1.5 years ago. As per the settings it looked like the slides were hidden on purpose based on devices starting from 800px and below.

File: static/js/components/FacultyCarousel.js (hunk 1 above)
Comment (arslanashraf7) [reply static/js/components/FacultyCarousel.js:27]: I updated it based on the media queries we usually use in the styling e.g. [this](https://github.com/mitodl/micromasters/blob/master/static/scss/program-page.scss#L417). And this slick setting will be applied on screens from `840px` and below until it reaches the lower threshold of `600px` In our case.

**Comment by arslanashraf7** [comment]
>1 slide for the tablet view is very big. Did you try 2?

@pdpinch 
Previously we were showing `2` slides for a breakpoint of `1200px` e.g. [reference](https://github.com/mitodl/micromasters/blob/master/static/js/components/FacultyCarousel.js#L37) and we were showing `1.5` slides for a breakpoint of `1000px` e.g. [reference](https://github.com/mitodl/micromasters/blob/master/static/js/components/FacultyCarousel.js#L29).

Based on the above reasons I went for 1 slide for tablets. I'll set it as 2 and try to see how it looks. I'll add a screenshot here.

**Comment by arslanashraf7** [comment]
@pdpinch I tried it with `2` and `1.5` slides options keeping the breakpoint at `840px`(Starting from 840px and ending at 600px). Attaching the screenshots below, let me know what you think.

**With 2 Slides**
Screenshot 2021-06-23 at 5 55 09 PM

**With 1.5 Slides**
Screenshot 2021-06-23 at 5 55 47 PM

File: static/js/components/FacultyCarousel.js (hunk 2 above)
Comment (pdpinch) [reply static/js/components/FacultyCarousel.js:24]: Huh. That's a mystery. Even if we wanted to hide it, this isn't the right way to do it. 

Can you do a before/after test with Lighthouse, to see what the penalty is for adding these back in?

**Comment by pdpinch** [comment]
Between the screenshots, I like 2 slides, but either is acceptable.

**Comment by arslanashraf7** [comment]
> Between the screenshots, I like 2 slides, but either is acceptable.

Sure, 2 slides it is then, PR updated.

File: static/js/components/FacultyCarousel.js (hunk 2 above)
Comment (arslanashraf7) [reply static/js/components/FacultyCarousel.js:24]: I did run a LightHouse performance check for mobile on the page locally and here are the results before and after adding the instructors on mobile. It didn't make a lot of difference on the performance though.
Note: I ran the test locally on a replica program page from RC, and with 3 instructors. Below are the results:

**On Master(No instructors on mobile)**
Mobile_performance_instructor_master

**On this branch (Instructors on mobile)**
Mobile_performance_with_instructors

**Comment by pdpinch** [comment]
Thanks Arslan. That is a horrible score, but it’s no worse than before. 

We have some work to do, but that should be in a separate issue.

**Comment by arslanashraf7** [comment]
> Thanks Arslan. That is a horrible score, but it’s no worse than before. We have some work to do, but that should be in a separate issue.

@pdpinch, Sure, Although the score on RC/Prod instances will be a lot better than this we can look at that later on.

@annagav I think this is ready for your review now.

**Review by annagav — APPROVED** [review]
Functionality looks good 👍

**Pull request merged**